Add a GitHub action for adding branch type labels

tomalec opened the pull request:
### Changes proposed in this Pull Request:

Sets `type: *` and `changelog: *` labels to a created PR, according to a branch name convention

It uses actions-ecosystem/action-add-labels to keep it consistent with 
https://github.com/woocommerce/woocommerce/blob/trunk/.github/workflows/update-feedback-labels.yml

### Screenshots:

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Pick a test repo
2. Follow the instructions in https://github.com/woocommerce/grow/compare/github-actions...add/label-action?expand=1#diff-ffef87c73e2a2202c94315b3bf7f301382b60802da7cb2db0e12935d936f8549


### Additional details:

1. This PR does follow the convention proposed in https://github.com/woocommerce/google-listings-and-ads/pull/1565. Therefore changes and unifies the set of branch prefixes and changelog entry types.
2. The flow proposed here, works well for regular single-package repos. However, it doesn't play well for monorepos, like this (`/grow`) one. I wonder how would we like to infer the package name to be set for [`[{package-name}] changelog: {type}`](https://github.com/woocommerce/grow/pull/18)? should we use branch names, like `actions/fix/123-slug`? Then `actions` or `github-actions` to follow the package name? Or maybe have another action to check the files modified, and set `package: storybook`, then let this action take that into consideration?
 @eason9487  

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

>

tomalec committed:
Ignore release branches in changelog.

tomalec committed:
Update the list of chagelog types,

to the one at https://github.com/woocommerce/google-listings-and-ads/pull/1565#issuecomment-1163223328

tomalec committed:
Add action to set PR labels

according to the branch name convention.
Add a reference in actions README.

eason9487 commented:
> should we use branch names, like `actions/fix/123-slug`? Then `actions` or `github-actions` to follow the package name? Or maybe have another action to check the files modified, and set `package: storybook`, then let this action take that into consideration?

The latter one seems to work. I'd prefer it if possible, so that we could remember fewer conventions. 👍

eason9487 commented on packages/js/github-actions/actions/branch-label/README.md:
```diff
@@ -0,0 +1,28 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set Labels
+        uses: woocommerce/grow/github-actions/branch-label@add/label-action
```
Note: this needs to be updated to `uses: woocommerce/grow/branch-label@actions-v1` before merging.

eason9487 commented on packages/js/github-actions/actions/branch-label/action.yml:
```diff
@@ -0,0 +1,45 @@
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v2
```
This step could be omitted since `actions-ecosystem/action-add-labels` can be run without the codebase.

eason9487 reviewed (approved):
Testing well. LGTM.

tomalec committed:
Remove not needed checkout from branch-label action

Addresses https://github.com/woocommerce/grow/pull/14#discussion_r905687861

tomalec committed:
Update branch-label action ref in readme,

to point to the released version.

Addresses https://github.com/woocommerce/grow/pull/14#discussion_r905686763
